system/lib/dlmalloc.c

```diff
@@ -713,6 +719,16 @@ defined(__i386__) || defined(__x86_64__))) ||                    \
 #define HAVE_MORECORE 1
 #endif  /* ONLY_MSPACES */
 #endif  /* HAVE_MORECORE */
+#ifndef UNSIGNED_MORECORE
+#define UNSIGNED_MORECORE 0
+#endif  /* UNSIGNED_MORECORE */
+#if UNSIGNED_MORECORE
+#define MORECORE_CANNOT_TRIM 1
+#else  /* !UNSIGNED_MORECORE */
+#ifndef MORECORE_CANNOT_TRIM
+#define MORECORE_CANNOT_TRIM 0
+#endif  /* MORECORE_CANNOT_TRIM */
```

Comment:
It looks like maybe if these three lines be remove we can continue to use `#ifndef MORECORE_CANNOT_TRIM` below to signify that it can trim?  (shinking the size of this patch?)

Reply:
You are right